RDKit 2023.03.2 breaks CI by no longer raising expected conformer generation error

**Describe the bug**

As of the 2023.03.2 release (https://github.com/rdkit/rdkit/pull/6379), RDKit can generate conformers for hexavalent octahedral molecules, meaning that our test case for handling conformer generation failures (using UF6) no longer raises the expected exception, [causing our CI to fail](https://github.com/openforcefield/openff-toolkit/actions/runs/5340124649/jobs/9679564928#step:15:369). 

```
________________ TestMolecule.test_conformer_generation_failure ________________
[gw1] linux -- Python 3.9.16 /home/runner/micromamba/envs/openff-toolkit-test/bin/python
openff/toolkit/tests/test_molecule.py:3679: in test_conformer_generation_failure
    molecule.generate_conformers(
E   Failed: DID NOT RAISE <class 'openff.toolkit.utils.exceptions.ConformerGenerationError'>
----------------------------- Captured stderr call -----------------------------
Warning: OEMMFFParams::PrepMol() : unable to type atom 1 U on residue UNL-1
Warning: : Force field setup failed due to missing parameters
Warning: OEMMFFParams::PrepMol() : unable to type atom 1 U on residue UNL-1
Warning: : Force field setup failed due to missing parameters
___________ TestRDKitToolkitWrapper.test_generate_conformers_failure ___________
[gw1] linux -- Python 3.9.16 /home/runner/micromamba/envs/openff-toolkit-test/bin/python
openff/toolkit/tests/test_toolkits.py:2862: in test_generate_conformers_failure
    toolkit.generate_conformers(molecule, n_conformers=1)
E   Failed: DID NOT RAISE <class 'openff.toolkit.utils.exceptions.ConformerGenerationError'>
```

--- a/docs/releasehistory.md
+++ b/docs/releasehistory.md
@@ -13,6 +13,8 @@ Releases follow the `major.minor.micro` scheme recommended by [PEP440](https://w
 ### Behavior changes
 
 ### Bugfixes
+- [PR #1654](https://github.com/openforcefield/openff-toolkit/pull/1654): Fixes issue [#1653](https://github.com/openforcefield/openff-toolkit/issues/1653), where a test that expected RDKit to fail began returning an error when RDKit became able to generate conformers for octahedral molecules. 
+
 
 ### New features
 

--- a/openff/toolkit/tests/test_molecule.py
+++ b/openff/toolkit/tests/test_molecule.py
@@ -3664,22 +3664,6 @@ class TestMolecule:
                 continue
             assert bond.is_in_ring()
 
-    @requires_rdkit
-    @requires_openeye
-    def test_conformer_generation_failure(self):
-        # This test seems possibly redundant, is it needed?
-        molecule = Molecule.from_smiles("F[U](F)(F)(F)(F)F")
-
-        with pytest.raises(ConformerGenerationError, match="Omega conf.*fail"):
-            molecule.generate_conformers(
-                n_conformers=1, toolkit_registry=OpenEyeToolkitWrapper()
-            )
-
-        with pytest.raises(ConformerGenerationError, match="RDKit conf.*fail"):
-            molecule.generate_conformers(
-                n_conformers=1, toolkit_registry=RDKitToolkitWrapper()
-            )
-
     def test_deepcopy_not_shallow(self):
         """
         Check that deep copies don't re-use any mutable data structures

--- a/openff/toolkit/tests/test_molecule.py
+++ b/openff/toolkit/tests/test_molecule.py
@@ -56,7 +56,6 @@ from openff.toolkit.topology.molecule import (
 )
 from openff.toolkit.utils import get_data_file_path
 from openff.toolkit.utils.exceptions import (
-    ConformerGenerationError,
     IncompatibleShapeError,
     IncompatibleTypeError,
     IncompatibleUnitError,

--- a/openff/toolkit/tests/test_toolkits.py
+++ b/openff/toolkit/tests/test_toolkits.py
@@ -1329,7 +1329,7 @@ class TestOpenEyeToolkitWrapper:
     def test_generate_conformers_failure(self):
         toolkit = OpenEyeToolkitWrapper()
 
-        molecule = Molecule.from_smiles("F[U](F)(F)(F)(F)F")
+        molecule = Molecule.from_smiles("C(C2)(C3)1CC23C1")
 
         with pytest.raises(ConformerGenerationError, match="Omega conf.*fail"):
             toolkit.generate_conformers(molecule, n_conformers=1)
@@ -2856,7 +2856,7 @@ class TestRDKitToolkitWrapper:
     def test_generate_conformers_failure(self):
         toolkit = RDKitToolkitWrapper()
 
-        molecule = Molecule.from_smiles("F[U](F)(F)(F)(F)F")
+        molecule = Molecule.from_smiles("C(C2)(C3)1CC23C1")
 
         with pytest.raises(ConformerGenerationError, match="RDKit conf.*fail"):
             toolkit.generate_conformers(molecule, n_conformers=1)

